File: pf-core/src/main/java/com/purdynet/util/PFColumnUtil.java
package com.purdynet.util;

import com.purdynet.graph.PFColumn;

import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: dnpurdy
 * Date: 8/31/13
 * Time: 3:05 PM
 * To change this template use File | Settings | File Templates.
 */
public class PFColumnUtil
{
    public static String getPattern(List<PFColumn> columns, Integer patternSize)
    {
        if(patternSize>columns.size()) return null;
        else
        {
            StringBuilder sb = new StringBuilder();
            for(int i = columns.size()-1-patternSize; i<columns.size(); i++)
            {
                sb.append(columns.get(i).toString());
                if(i!=columns.size()-1) sb.append("-");
            }
            return sb.toString();
        }
    }
}

File: pf-core/src/main/scala/com/purdynet/util/PFColumnUtil.scala
package com.purdynet.util

import com.purdynet.graph.PFColumn

import scala.collection.JavaConverters._
/**
  * Created by dnpurdy on 11/2/16.
  */
object PFColumnUtil {
  def getPattern(columnsJ: java.util.List[PFColumn], patternSize: Integer): String = {
    val columns = columnsJ.asScala
    if (patternSize > columns.size) null
    else {
      val sb: StringBuilder = new StringBuilder
      var i: Int = columns.size - 1 - patternSize
      while (i < columns.size) {
        {
          sb.append(columns(i).toString)
          if (i != columns.size - 1) sb.append("-")
        }
        {
          i += 1; i - 1
        }
      }
      sb.toString
    }
  }
}
